StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1401FieldsMustBePrivate.cs

```diff
@@ -49,14 +50,15 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            context.RegisterSymbolAction(this.AnalyzeField, SymbolKind.Field);
+            context.RegisterSymbolAction(AnalyzeField, SymbolKind.Field);
         }
 
-        private void AnalyzeField(SymbolAnalysisContext symbolAnalysisContext)
+        private static void AnalyzeField(SymbolAnalysisContext symbolAnalysisContext)
         {
             var fieldDeclarationSyntax = (IFieldSymbol)symbolAnalysisContext.Symbol;
-            if (!this.IsFieldPrivate(fieldDeclarationSyntax) &&
-                this.IsParentAClass(fieldDeclarationSyntax) &&
+            if (!IsFieldPrivate(fieldDeclarationSyntax) &&
+                !IsStaticReadonly(fieldDeclarationSyntax) &&
+                IsParentAClass(fieldDeclarationSyntax) &&
```

Comment:
I have no clue, I didn't make the initial implementation, just added the static readonly stuff.